Marlin/temperature.cpp

```diff
@@ -1746,15 +1746,15 @@ void Temperature::isr() {
 
       #if ENABLED(FAN_SOFT_PWM)
         #if HAS_FAN0
-          soft_pwm_count_fan[0] = (soft_pwm_count_fan[0] & pwm_mask) + soft_pwm_amount_fan[0] >> 1;
+          soft_pwm_count_fan[0] = ((soft_pwm_count_fan[0] & pwm_mask) + soft_pwm_amount_fan[0]) >> 1;
```

Comment:
It would be a good idea to check with @StefanBruens before making this kind of change. Here is the original commit with dithering implemented.

https://github.com/MarlinFirmware/Marlin/commit/35a55d5757c1ed415cd5c9b4225f5c2011f0fbf1#diff-7e90b29c765c3047c7dc2aa8eff0856cR1582

Reply:
I believe that this change is purely cosmetic because addition and subtraction operators have presence over bitwise left shift and right shift operators ie (x+y>>1 == (x+y) >> 1).

Reply:
Holy crap, you're right. It's one of those aspects of C++ operator precedence that makes me shake my head.

Reply:
Actually the current implementation is wrong, then. The ">> 1" scaling should only be applied to the fan speed value, not the remainder.
The bug is not critical (resolution is halved), but should be corrected anyway.